apis/placement/v1beta1/resourcesnapshot_types.go

```diff
@@ -39,9 +42,8 @@ const (
 // Its spec is immutable.
 // We may need to produce more than one resourceSnapshot for all the resources a ResourcePlacement selected to get around the 1MB size limit of k8s objects.
 // We assign an ever-increasing index for each such group of resourceSnapshots.
-// The name convention of a clusterResourceSnapshot is {CRPName}-{resourceIndex}(-{subindex})*
-// where the name of the first snapshot of a group has no subindex part so its name is {CRPName}-{resourceIndex}.
-// resourceIndex will begin with 0.
+// The name convention of a clusterResourceSnapshot is {CRPName}-{resourceIndex}-{subindex} where resourceIndex and
+// subindex will begin with 0.
```

Comment:
I guess this is to avoid conflict between different resource snapshot's name? Obviously this is not ideal. What are our other options?

Reply:
I get that restricting CRP name to not having "-" is not ideal either.

Reply:
The purpose of renaming the first snapshot is to keep snapshot names consistent no matter it's a sub-snapshot or not as mentioned in the "special note" of the PR description in order to avoid special handling in some cases.

To solve the duplication of snapshots (policySnapshot/resourceSnapshot), i'm thinking when we create a snapshot, we create a CRP + {random-suffix} as the a new CRP name and store it in each snapshot. The policySnapshot name becomes {CRPNameWithRandomSuffix}- {index} and resourceSnapshot name becomes {CRPNameWithRandomSuffix}-{resourceIndex}-{subIndex}. The CRPNameWithRandomSuffix should be unchanged during the CRP lifecycle.

WDYT?